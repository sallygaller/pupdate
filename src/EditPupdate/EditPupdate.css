.EditPupdate h2 {
    text-align: center;
}

.EditPupdate {
    border-radius: 5px;
    background-color: #f2f2f2;
    padding: 20px;  
}

.EditPupdate-form {
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    width: 30%;
}

.EditPupdate textarea {
    font-family: sans-serif
}

.EditPupdate input {
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    display: inline-block;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
} 

.EditPupdate label {
    margin-bottom: 5px;
}

.EditPupdate button {
    width: 100%;
    border: #F2AD1D solid 2px;
    border-radius: 25px;
    padding: 14px 20px;
    margin: 8px 0;
    cursor: pointer;
}

.EditPupdate button:hover {
    background-color: #F2AD1D;
}

.EditPupdate-error {
    color: red;
}