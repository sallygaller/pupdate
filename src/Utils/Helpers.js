export default function PupSize(size) {
  let sizeText = "";
  if (size === "XS") {
    sizeText = "Extra Small (under 10lbs)";
  } else if (size === "S") {
    sizeText = "Small (10-30lbs)";
  } else if (size === "M") {
    sizeText = "Medium (30-60lbs)";
  } else if (size === "L") {
    sizeText = "Large (60-90lbs)";
  } else if (size === "XL") {
    sizeText = "Extra Large (over 90lbs)";
  }
  return sizeText;
}
