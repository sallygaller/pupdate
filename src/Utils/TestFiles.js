export const pupdates = [
  {
    id: 1,
    date: "2020-03-02T08:00:00.000Z",
    starttime: "09:30:00",
    endtime: "10:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park!",
    organizer: 1,
  },
  {
    id: 2,
    date: "2020-03-01T08:00:00.000Z",
    starttime: "15:30:00",
    endtime: "17:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park! I'll bring a frisbee",
    organizer: 1,
  },
];

export const pups = [
  {
    id: 1,
    name: "Eddie",
    breed: "dachshund",
    mix: true,
    age: "Adult",
    size: "M",
    nervous: true,
    rambunctious: false,
    gentle: true,
    wrestling: false,
    walks: true,
    parks: false,
    foodobsessed: true,
    ballobsessed: false,
    description:
      "Eddie can be a bit shy around other dogs at first, but he is really sweet! He loves going on long walks where there is a lot to sniff",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 1,
  },
  {
    id: 2,
    name: "Buddy",
    breed: "beagle",
    mix: true,
    age: "Adult",
    size: "L",
    nervous: false,
    rambunctious: true,
    gentle: false,
    wrestling: true,
    walks: false,
    parks: true,
    foodobsessed: false,
    ballobsessed: false,
    description:
      "Buddy can be a bit full on for some dogs, but he really just loves to play! He especially loves puppies",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 2,
  },
];
