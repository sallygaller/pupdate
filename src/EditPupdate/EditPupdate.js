import React, { useState, useEffect } from "react";
import { useHistory } from "react-router";
import TokenService from "../services/token-service";
import { API_ENDPOINT } from "../config";
import SearchLocationInput from "../SearchLocation/SearchLocation";
import moment from "moment";
import "./EditPupdate.css";

export default function EditPupdate(props) {
  const [error, setError] = useState(null);
  const [id, setId] = useState("");
  const [date, setDate] = useState("");
  const [starttime, setStarttime] = useState("");
  const [endtime, setEndtime] = useState("");
  const [location, setLocation] = useState("");
  const [description, setDescription] = useState("");
  const history = useHistory();

  useEffect(() => {
    const { pupdateId } = props.match.params;
    fetch(API_ENDPOINT + `/pupdates/${pupdateId}`, {
      method: "GET",
      headers: {
        authorization: `Bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then((res) => {
        if (!res.ok) return res.json().then((error) => Promise.reject(error));
        return res.json();
      })
      .then((responseData) => {
        setId(responseData.id);
        setDate(responseData.date);
        setStarttime(responseData.starttime);
        setEndtime(responseData.endtime);
        setLocation(responseData.location);
        setDescription(responseData.description);
      })
      .catch((error) => {
        console.error(error);
        this.setState({ error });
      });
  }, []);

  const handleSubmit = (e) => {
    e.preventDefault();
    const pupdate = {
      id,
      date,
      starttime,
      endtime,
      location,
      description,
    };
    fetch(API_ENDPOINT + `/pupdates/${pupdate.id}`, {
      method: "PATCH",
      body: JSON.stringify(pupdate),
      headers: {
        "content-type": "application/json",
        authorization: `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then((res) => {
        if (!res.ok) return res.json().then((error) => Promise.reject(error));
      })
      .then(() => {
        history.push("/pupdates");
      })
      .catch((error) => {
        setError(error.message);
      });
  };

  return (
    <div className="EditPupdate">
      <h2>Edit pupdate</h2>
      <form className="EditPupdate-form" onSubmit={(e) => handleSubmit(e)}>
        <div className="EditPupdate-error" role="alert">
          {error && (
            <p>
              {error.message}
              {error}
            </p>
          )}
        </div>
        <label htmlFor="location">Location:</label>
        <SearchLocationInput
          location={location}
          setLocation={setLocation}
          placeholder={"Enter a location (e.g. a local park)"}
        />
        <label htmlFor="date">Date: </label>
        <input
          type="date"
          value={moment(date).format("YYYY-MM-DD")}
          onChange={(e) => setDate(e.target.value)}
        ></input>
        <label htmlFor="startTime">Start Time: </label>
        <input
          type="time"
          value={starttime}
          onChange={(e) => setStarttime(e.target.value)}
        ></input>
        <label htmlFor="endTime">End Time: </label>
        <input
          type="time"
          value={endtime}
          onChange={(e) => setEndtime(e.target.value)}
        ></input>
        <label htmlFor="description">Description:</label>
        <textarea
          id="description"
          name="description"
          className="EditPupdate-textarea"
          rows="5"
          cols="10"
          value={description}
          onChange={(e) => setDescription(e.target.value)}
        ></textarea>
        <button type="submit">Submit</button>
        <button type="button">Cancel</button>
      </form>
    </div>
  );
}
