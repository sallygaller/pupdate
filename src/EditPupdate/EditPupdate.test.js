import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import { pupdates } from "../Utils/TestFiles";
import renderer from "react-test-renderer";
import EditPupdate from "./EditPupdate";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <EditPupdate
        name="EditPupdate"
        pupdate={pupdates[0]}
        match={{ params: { id: 1 }, isExact: true, path: "", url: "" }}
      />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <EditPupdate
        name="EditPupdate"
        pupdate={pupdates[0]}
        match={{ params: { id: 1 }, isExact: true, path: "", url: "" }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
