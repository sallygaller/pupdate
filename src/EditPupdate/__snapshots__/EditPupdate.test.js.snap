// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="EditPupdate"
>
  <h2>
    Edit Pupdate
  </h2>
  <form
    className="EditPupdate-form"
    onSubmit={[Function]}
  >
    <div
      className="EditPupdate-error"
      role="alert"
    />
    <label
      htmlFor="location"
    >
      Location:
    </label>
    <div
      className="search-location-input"
    >
      <input
        onChange={[Function]}
        placeholder="Enter a location (e.g. a local park)"
        value=""
      />
    </div>
    <label
      htmlFor="date"
    >
      Date: 
    </label>
    <input
      onChange={[Function]}
      type="date"
      value=""
    />
    <label
      htmlFor="startTime"
    >
      Start Time: 
    </label>
    <input
      onChange={[Function]}
      type="time"
      value=""
    />
    <label
      htmlFor="endTime"
    >
      End Time: 
    </label>
    <input
      onChange={[Function]}
      type="time"
      value=""
    />
    <label
      htmlFor="description"
    >
      Description:
    </label>
    <textarea
      className="EditPupdate-textarea"
      cols="10"
      id="description"
      name="description"
      onChange={[Function]}
      rows="5"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </form>
</div>
`;
